silver_payu/payment_processors.py

```diff
@@ -142,28 +134,43 @@ def _charge_transaction(self, transaction):
 
         payment = TokenPayment(payment_details, token)
 
-        # TODO: handle connection failing
-        result = payment.pay()
+        try:
+            result = payment.pay()
+        except Exception as error:
+            transaction.fail(fail_reason='Connectivity issues between server '
+                                         'and PayU gateway')
+            transaction.save()
+            return False
 
         return self._parse_result(transaction, result)
 
     def _parse_result(self, transaction, result):
         try:
-            # TODO: return json from django-payu-ro
-            if not int(json.loads(result)["code"]):
-                self.update_transaction_status(transaction, "pending")
+            result = json.loads(result)
+
+            if not result["code"]:
+                transaction.process()
+                transaction.save()
                 return True
             else:
-                self.update_transaction_status(transaction, "failed")
-                transaction.data["result"] = result
+                error_code, error_reason = self._parse_error(result)
+                transaction.fail(fail_code=error_code, fail_reason=error_reason)
         except Exception as e:
```

Comment:
This might hide transactions that might pass in the future. There shouldn't be a catch all exception here.